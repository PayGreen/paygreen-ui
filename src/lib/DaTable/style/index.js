import styled from 'styled-components';
import { responsiveSpaces } from '../../../shared/spaces';

const DaTableBase = styled.div`
    ${responsiveSpaces('margin')};

    @media ${props => props.theme.screen.max.lg} {
        max-width: ${props => props.theme.blockWidth[props.blockWidth]};
    }

    @media ${props => props.theme.screen.min.lg} {
        width: fit-content;
    }

    .table {
        @media ${props => props.theme.screen.min.lg} {
            display: table;
            border-collapse: collapse;
        }
    }

    .noRowMessage {
        background-color: ${props => props.theme.wab.white10};
        color: ${props => props.theme.wab.grey60};

        @media ${props => props.theme.screen.max.lg} {
            border-radius: ${props => props.theme.radius.sm};
            margin: ${props => props.theme.space.xs};
            padding: ${props => props.theme.space.sm};
        }

        @media ${props => props.theme.screen.min.lg} {
            padding: ${props => props.theme.space.md};
        }
    }
`;

export { DaTableBase };
