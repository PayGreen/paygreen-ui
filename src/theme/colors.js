const mainColors = {
    green: '#33ad73',
    blue: '#00a99d',
    orange: '#e58a28',
    pink: '#ec6a6a',
    purple: '#a26ae1',
    red: '#b54025',
};

const colors = {
    green: {
        main: mainColors.green,
        gradientBase: '#77d1a6',
        gradientShade: '#a4db95',
    },
    blue: {
        main: mainColors.blue,
        gradientBase: '#51c6be',
        gradientShade: '#76bbd8',
    },
    orange: {
        main: mainColors.orange,
        gradientBase: '#fab56b',
        gradientShade: '#faa272',
    },
    pink: {
        main: mainColors.pink,
        gradientBase: '#e98e90',
        gradientShade: '#e98eb9',
    },
    purple: {
        main: mainColors.purple,
        gradientBase: '#b887ee',
        gradientShade: '#dd94ef',
    },
};

export { mainColors, colors };
