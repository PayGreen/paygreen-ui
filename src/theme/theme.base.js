const responsive = {
    sm: 550,
    md: 960,
    lg: 1200,
    xl: 1360,
};

const query = (value, operator) => {
    if (operator == 'max') {
        value -= 1;
    }
    return operator + '-width: ' + value + 'px';
};

const pallet = {
    green: {
        main: '#33ad73',
        light: '#d5f3e5',
    },
    blue: {
        main: '#00a99d',
        light: '#e1f3f2',
    },
    orange: {
        main: '#e58a28',
        light: '#fbe6ce',
    },
};

export const ThemeBase = {
    name: 'Base',
    color: {
        primary: {
            ...pallet.green,
            gradientBase: '#77d1a6',
            gradientShade: '#a4db95',
        },
        secondary: {
            ...pallet.blue,
            gradientBase: '#51c6be',
            gradientShade: '#76bbd8',
        },
        tertiary: {
            ...pallet.orange,
            gradientBase: '#fab56b',
            gradientShade: '#faa272',
        },
        quaternary: {
            main: '#ec6a6a',
            light: '#ffe0e1',
            gradientBase: '#e98e90',
            gradientShade: '#e98eb9',
        },
        quinary: {
            main: '#a26ae1',
            light: '#e9d9fb',
            gradientBase: '#b887ee',
            gradientShade: '#dd94ef',
        },
    },
    status: {
        success: {
            ...pallet.green,
        },
        default: {
            ...pallet.blue,
        },
        warning: {
            ...pallet.orange,
        },
        danger: {
            main: '#b54025',
            light: '#f7d7d0',
        },
    },
    font: {
        size: {
            tiny: '1rem',
            xxs: '1.2rem',
            xs: '1.4rem',
            sm: '1.6rem',
            base: '1.8rem',
            md: '2.1rem',
            lg: '3rem',
            xl: '4rem',
        },
        weight: {
            normal: 400,
            bold: 700,
        },
        lineHeight: {
            base: 1,
            md: 1.5,
        },
        spacing: '0.3em',
        underline: {
            space: {
                tiny: '8px',
                xxs: '10px',
                xs: '10px',
                sm: '12px',
                base: '12px',
                md: '12px',
                lg: '24px',
                xl: '24px',
            },
            line: {
                tiny: '2px',
                xxs: '2px',
                xs: '2px',
                sm: '2px',
                base: '2px',
                md: '2px',
                lg: '3px',
                xl: '3px',
            },
            width: {
                tiny: '30px',
                xxs: '40px',
                xs: '40px',
                sm: '50px',
                base: '50px',
                md: '65px',
                lg: '80px',
                xl: '80px',
            },
        },
    },
    line: '2px',
    dash: '3px',
    radius: {
        none: '0',
        sm: '4px',
        md: '8px',
        lg: '12px',
    },
    arrow: {
        sm: '8px',
        md: '12px',
        lg: '16px',
    },
    space: {
        none: '0',
        xs: '4px',
        sm: '8px',
        md: '16px',
        lg: '32px',
        xl: '64px',
    },
    grid: {
        header: '45px',
        headerBig: '70px',
        topbar: '45px',
        sidebar: '380px',
        maxWidth: '1400px',
    },
    blockShift: {
        none: '0',
        xs: '64px',
        sm: '128px',
        md: '256px',
        lg: '320px',
        xl: '384px',
    },
    blockSpace: {
        none: '0',
        xs: '16px',
        sm: '24px',
        md: '40px',
        lg: '52px',
        xl: '68px',
    },
    blockWidth: {
        sm: '400px',
        md: '560px',
        lg: '640px',
        xl: '800px',
    },
    blockHeader: {
        sm: '140px',
        md: '180px',
        lg: '210px',
        xl: '240px',
    },
    bannerHeight: {
        xs: '33vh',
        sm: '50vh',
        lg: '100vh',
    },
    wave: {
        simple: '8vw',
        double: '11vw',
    },
    icon: {
        size: {
            xxs: '12px',
            xs: '16px',
            sm: '20px',
            md: '24px',
            lg: '32px',
            xl: '40px',
        },
        shift: {
            xxs: '2px',
            xs: '3px',
            sm: '4px',
            md: '4px',
            lg: '6px',
            xl: '6px',
        },
    },
    imageSize: {
        auto: 'auto',
        xxs: '150px',
        xs: '240px',
        sm: '400px',
        md: '560px',
        lg: '640px',
        xl: '760px',
    },
    imageCircleSize: {
        auto: 'auto',
        xxs: '38px',
        xs: '60px',
        sm: '80px',
        md: '100px',
        lg: '120px',
        xl: '140px',
    },
    logoHeight: {
        none: 'auto',
        xs: '20px',
        sm: '30px',
        md: '40px',
        lg: '50px',
        xl: '60px',
    },
    logoWidth: {
        none: 'auto',
        xs: '90px',
        sm: '120px',
        md: '150px',
        lg: '200px',
        xl: '250px',
    },
    loaderSize: {
        xxs: '40px',
        xs: '50px',
        sm: '60px',
        md: '70px',
        lg: '80px',
        xl: '90px',
    },
    transition: {
        xxs: '0.1s',
        xs: '0.2s',
        sm: '0.3s',
        md: '0.5s',
        lg: '0.7s',
        xl: '1s',
    },
    zindex: {
        hidden: -100,
        layer: -5,
        background: -1,
        base: 0,
        content: 1,
        corner: 3,
        front: 5,
        loader: 90,
        subHeader: 100,
        menu: 105,
        header: 110,
        popin: 115,
        modal: 120,
    },
    query: {
        min: {
            sm: query(responsive.sm, 'min'),
            md: query(responsive.md, 'min'),
            lg: query(responsive.lg, 'min'),
            xl: query(responsive.xl, 'min'),
        },
        max: {
            sm: query(responsive.sm, 'max'),
            md: query(responsive.md, 'max'),
            lg: query(responsive.lg, 'max'),
            xl: query(responsive.xl, 'max'),
        },
    },
    button: {
        paddingWidth: {
            sm: '22px',
            md: '26px',
            lg: '30px',
        },
        paddingHeight: {
            sm: '12px',
            md: '14px',
            lg: '16px',
        },
        shift: '3px',
        font: {
            sm: '1.2rem',
            md: '1.4rem',
            lg: '1.5rem',
        },
        letterSpacing: {
            sm: '0.3rem',
            md: '0.4rem',
            lg: '0.45rem',
        },
    },
    daButton: {
        buttonHeight: {
            sm: '32px',
            md: '38px',
            lg: '44px',
        },
        shift: {
            sm: '2px',
            md: '4px',
            lg: '6px',
        },
        font: {
            sm: '1.35rem',
            md: '1.4rem',
            lg: '1.7rem',
        },
        iconSpacing: {
            sm: '0.9rem',
            md: '1rem',
            lg: '1.1rem',
        },
    },
    form: {
        inputWidth: {
            xs: '260px',
            sm: '330px',
            md: '400px',
            lg: '500px',
        },
        textareaHeight: '200px',
        radio: {
            md: '120px',
            lg: '150px',
        },
        checkboxSize: {
            sm: '18px',
            md: '20px',
            lg: '22px',
        },
        radioButtonSize: {
            sm: '19px',
            md: '21px',
            lg: '23px',
        },
        radioButtonSpace: {
            sm: '8px',
            md: '10px',
            lg: '10px',
        },
        toggle: '30px',
    },
    corner: {
        size: {
            lg: '34px',
        },
        padding: {
            lg: '6px',
        },
    },
    shadow: {
        size: {
            none: '0 0 0',
            xs: '1px 1px 5px',
            sm: '1px 1px 10px',
            md: '2px 2px 15px',
            lg: '3px 3px 20px',
        },
        opacity: {
            none: 1,
            xs: 0.95,
            sm: 0.9,
            md: 0.8,
            lg: 0.7,
        },
    },
    rotateSize: {
        d0: '0deg',
        d45: '45deg',
        d90: '90deg',
        d135: '135deg',
        d180: '180deg',
        d225: '225deg',
        d270: '270deg',
        d315: '315deg',
    },
    dataBar: {
        height: '10px',
        width: {
            sm: '300px',
            md: '400px',
            lg: '500px',
            xl: '600px',
        },
    },
};
