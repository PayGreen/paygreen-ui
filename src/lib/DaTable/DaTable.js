import React from 'react';
import PropTypes from 'prop-types';
import { spaceOptions, spaceDefault } from '../../shared/constants';
import Checkbox from '../Checkbox/Checkbox';
import DaTableHeadCell from '../DaTableHeadCell/DaTableHeadCell';
import DaTableHead from '../DaTableHead/DaTableHead';
import DaTableBody from '../DaTableBody/DaTableBody';
import DaTableRow from '../DaTableRow/DaTableRow';
import { DaTableBase } from './style';

const DaTable = ({
    children,
    isLoading,
    loadingRowNumber,
    loadingMainCellNumber,
    ...rest
}) => {
    let columnNumber = 0;
    const loadingCheckboxCellIndex = [];
    let noRowMessage = null;
    let hasRow = true;

    if (isLoading) {
        React.Children.map(children, child => {
            if (
                typeof child === 'object' &&
                (child.type === DaTableHead ||
                    child.props.className === 'DaTableHead')
            ) {
                columnNumber = child.props.children.length;

                React.Children.map(child.props.children, (headCell, index) => {
                    if (
                        typeof headCell === 'object' &&
                        (headCell.type === DaTableHeadCell ||
                            headCell.props.className === 'DaTableHeadCell') &&
                        (headCell.props.isCheckbox ||
                            (headCell.props.children &&
                                headCell.props.children.type === Checkbox))
                    ) {
                        loadingCheckboxCellIndex.push(index);
                    }
                });
            }
        });
    } else {
        React.Children.map(children, child => {
            if (
                typeof child === 'object' &&
                (child.type === DaTableBody ||
                    child.props.className === 'DaTableBody')
            ) {
                hasRow = false;

                React.Children.map(child.props.children, row => {
                    if (
                        typeof row === 'object' &&
                        (row.type === DaTableRow ||
                            row.props.className === 'DaTableRow')
                    ) {
                        hasRow = true;
                    }
                });

                if (!hasRow) {
                    noRowMessage = child.props.children;
                }
            }
        });
    }

    return (
        <DaTableBase {...rest}>
            <div className="table">
                {React.Children.map(children, child => {
                    if (!child) {
                        return null;
                    } else if (
                        child.type === DaTableBody ||
                        child.props.className === 'DaTableBody'
                    ) {
                        return React.cloneElement(child, {
                            hasRow: hasRow,
                            isLoading: isLoading,
                            loadingColumnNumber: columnNumber,
                            loadingRowNumber: loadingRowNumber,
                            loadingMainCellNumber: loadingMainCellNumber,
                            loadingCheckboxCellIndex: loadingCheckboxCellIndex,
                        });
                    } else {
                        return child;
                    }
                })}
            </div>

            {!hasRow ? (
                <div className="noRowMessage">
                    {noRowMessage ? noRowMessage : 'No data'}
                </div>
            ) : null}
        </DaTableBase>
    );
};

DaTable.propTypes = {
    isLoading: PropTypes.bool,
    loadingRowNumber: PropTypes.number,
    loadingMainCellNumber: PropTypes.number,
    blockWidth: PropTypes.oneOf(Object.values(spaceOptions)),
    marginLateral: PropTypes.oneOf(Object.values(spaceOptions)),
    marginTop: PropTypes.oneOf(Object.values(spaceOptions)),
    marginBottom: PropTypes.oneOf(Object.values(spaceOptions)),
};

DaTable.defaultProps = {
    isLoading: false,
    loadingRowNumber: 10,
    loadingMainCellNumber: 4,
    blockWidth: spaceOptions.md,
    marginLateral: spaceDefault,
    marginTop: spaceDefault,
    marginBottom: spaceDefault,
};

export default DaTable;
