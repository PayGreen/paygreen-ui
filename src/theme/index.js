import { ThemeDefault } from './theme.default';
import { ThemeDark } from './theme.dark';

export { ThemeDefault, ThemeDark };
