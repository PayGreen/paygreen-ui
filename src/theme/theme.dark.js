import patternDark from './assets/pattern-dark.png';
import { baseDark, gradients } from './colors';
import { ThemeBase } from './theme.base';

const ThemeDark = {
    ...ThemeBase,
    name: 'Dark',
    color: {
        primary: {
            ...baseDark.green,
            ...gradients.green,
        },
        secondary: {
            ...baseDark.blue,
            ...gradients.blue,
        },
        tertiary: {
            ...baseDark.orange,
            ...gradients.orange,
        },
        quaternary: {
            ...baseDark.pink,
            ...gradients.pink,
        },
        quinary: {
            ...baseDark.purple,
            ...gradients.purple,
        },
    },
    status: {
        success: {
            ...baseDark.green,
        },
        default: {
            ...baseDark.blue,
        },
        warning: {
            ...baseDark.orange,
        },
        danger: {
            ...baseDark.red,
        },
    },
    wab: {
        white00: '#202020',
        white10: '#2c2c2c',
        white20: '#383838',

        grey10: '#494949',
        grey20: '#5b5b5b',
        grey30: '#7f7f7f',
        grey40: '#a2a2a2',
        grey50: '#cfcfcf',
        grey60: '#eeeeee',

        black00: '#ffffff',
    },
    pattern: patternDark,
    logoColor: '#ffffff',
};

export { ThemeDark };
