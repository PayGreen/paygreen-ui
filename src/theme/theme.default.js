import { ThemeBase } from './theme.base';
import patternLight from './assets/pattern-light.png';

let ThemeDefault = { ...ThemeBase };
ThemeDefault = Object.assign(ThemeDefault, {
    name: 'Default',
    wab: {
        white00: '#ffffff',
        white10: '#fafafa',
        white20: '#f6f6f6',

        grey10: '#eeeeee',
        grey20: '#dbdbdb',
        grey30: '#aaaaaa',
        grey40: '#858585',
        grey50: '#636363',
        grey60: '#434343',

        black00: '#000000',
    },
    pattern: patternLight,
});

export { ThemeDefault };
