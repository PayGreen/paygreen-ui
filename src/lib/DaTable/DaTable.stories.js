import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs, boolean, select } from '@storybook/addon-knobs';
import {
    folder,
    formStatusOptions,
    colorPalletOptions,
    fontSizeOptions,
    iconSizeOptions,
    spaceOptions,
} from '../../shared/constants';
import {
    CheckBoldIcon,
    CrossBoldIcon,
    ClockBoldIcon,
    ArrowBottomIcon,
    MenuIcon,
} from '../Icon/Icon';
import Text from '../Text/Text';
import DaTableCell from '../DaTableCell/DaTableCell';
import DaTableRow from '../DaTableRow/DaTableRow';
import DaTableBody from '../DaTableBody/DaTableBody';
import DaTableHeadCell from '../DaTableHeadCell/DaTableHeadCell';
import DaTableHead from '../DaTableHead/DaTableHead';
import DaInput from '../DaInput/DaInput';
import DaTable from './DaTable';

const iconStyles = {
    iconSize: iconSizeOptions.xs,
    colorPallet: colorPalletOptions.status,
    marginRight: spaceOptions.xs,
};

const textStyles = {
    textSize: fontSizeOptions.xs,
    colorPallet: colorPalletOptions.status,
};

const status = {
    icon: {
        validated: (
            <CheckBoldIcon
                {...iconStyles}
                colorStatus={formStatusOptions.success}
            />
        ),
        waiting: (
            <ClockBoldIcon
                {...iconStyles}
                colorStatus={formStatusOptions.warning}
            />
        ),
        refused: (
            <CrossBoldIcon
                {...iconStyles}
                colorStatus={formStatusOptions.danger}
            />
        ),
    },
    text: {
        validated: (
            <Text {...textStyles} colorStatus={formStatusOptions.success}>
                Validated
            </Text>
        ),
        waiting: (
            <Text {...textStyles} colorStatus={formStatusOptions.warning}>
                Waiting
            </Text>
        ),
        refused: (
            <Text {...textStyles} colorStatus={formStatusOptions.danger}>
                Refused
            </Text>
        ),
    },
};

const sampleRows = [
    {
        date: '27/05/2020',
        name: 'Laurie Fils',
        amount: '341.90',
        type: 'Cash',
        status: 'waiting',
    },
    {
        date: '26/05/2020',
        name: 'Matt Uritay',
        amount: '23.41',
        type: 'Recurring',
        status: 'waiting',
    },
    {
        date: '25/05/2020',
        name: 'Pedro Madère',
        amount: '12.10',
        type: 'Cash',
        status: 'waiting',
    },
    {
        date: '24/05/2020',
        name: 'Alex Pert',
        amount: '70.85',
        type: 'Tokenize',
        status: 'validated',
    },
    {
        date: '23/05/2020',
        name: 'Benja Mince',
        amount: '54.50',
        type: 'Cash',
        status: 'validated',
    },
    {
        date: '22/05/2020',
        name: 'Benoît de Cajou',
        amount: '93.00',
        type: 'Cash',
        status: 'validated',
    },
    {
        date: '21/05/2020',
        name: 'Guy Home',
        amount: '9.25',
        type: 'Cash',
        status: 'validated',
    },
    {
        date: '20/05/2020',
        name: 'Raph Histolé',
        amount: '89.92',
        type: 'Tokenize',
        status: 'refused',
    },
    {
        date: '19/05/2020',
        name: 'Fa Nini',
        amount: '34.60',
        type: 'Recurring',
        status: 'validated',
    },
    {
        date: '18/05/2020',
        name: 'Pierrick Aimor-Tis',
        amount: '71.05',
        type: 'Cash',
        status: 'refused',
    },
];

const isActiveLabel = 'First line active';

storiesOf(folder.tables + folder.sub.daTable + 'DaTable', module)
    .addDecorator(withKnobs)
    .add(
        'DaTable',
        () => (
            <DaTable
                marginLateral={select(
                    'Lateral margins',
                    spaceOptions,
                    spaceOptions.sm,
                )}
                marginTop={select('Margin top', spaceOptions, spaceOptions.sm)}
                marginBottom={select(
                    'Margin bottom',
                    spaceOptions,
                    spaceOptions.sm,
                )}
            >
                <DaTableHead resultsLabel="10 results">
                    <DaTableHeadCell
                        isCheckbox={true}
                        label="Select/deselect all"
                    >
                        <input type="checkbox" style={{ display: 'block' }} />
                    </DaTableHeadCell>

                    <DaTableHeadCell label="ID" />

                    <DaTableHeadCell
                        sortIcon={<ArrowBottomIcon title="Sort DESC on Date" />}
                        label="Date"
                    />

                    <DaTableHeadCell label="Name">
                        <DaInput placeholder="Search" fieldSize="sm" />
                    </DaTableHeadCell>

                    <DaTableHeadCell
                        sortIcon={
                            <ArrowBottomIcon
                                isActive={true}
                                title="Sort ASC on Amount"
                            />
                        }
                        label="Amount"
                    />

                    <DaTableHeadCell
                        groupIcon={<MenuIcon title="Group by Type" />}
                        label="Type"
                    />

                    <DaTableHeadCell
                        groupIcon={<MenuIcon title="Group by Status" />}
                        label="Status"
                    />
                </DaTableHead>

                <DaTableBody>
                    {sampleRows.map((sample, index) => (
                        <DaTableRow
                            key={index}
                            isActive={
                                !index ? boolean(isActiveLabel, false) : false
                            }
                        >
                            <DaTableCell isCheckbox={true}>
                                <input
                                    type="checkbox"
                                    checked={
                                        !index
                                            ? boolean(isActiveLabel, false)
                                            : false
                                    }
                                    readOnly={true}
                                />
                            </DaTableCell>

                            <DaTableCell isId={true}>
                                {3400 + index}
                            </DaTableCell>

                            <DaTableCell isMain={false} label="Date">
                                {sample.date}
                            </DaTableCell>

                            <DaTableCell>{sample.name}</DaTableCell>

                            <DaTableCell>{sample.amount}&nbsp;€</DaTableCell>

                            <DaTableCell isMain={false} label="Type">
                                {sample.type}
                            </DaTableCell>

                            <DaTableCell isMain={false} label="Status">
                                {status.icon[sample.status]}
                                {status.text[sample.status]}
                            </DaTableCell>
                        </DaTableRow>
                    ))}
                </DaTableBody>
            </DaTable>
        ),
        {
            notes:
                'input search is not functional, it is used here only for layout purpose',
        },
    );
