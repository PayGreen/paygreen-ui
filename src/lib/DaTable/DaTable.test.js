import React from 'react';
import TestRenderer from 'react-test-renderer';
import { ThemeDefault } from '../../theme';
import DaTable from './DaTable';

it('renders without crashing', () => {
    const component = TestRenderer.create(
        <DaTable theme={ThemeDefault}>
            Something
        </DaTable>
    );
    expect(component.toJSON()).toMatchSnapshot();
});