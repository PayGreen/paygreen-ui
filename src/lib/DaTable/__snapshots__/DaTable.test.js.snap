// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="sc-htpNat efseZc"
>
  <div
    className="sc-bwzfXH iBHjTe"
  >
    <div
      className="sc-bdVaJa fCCeHr cell-checkbox"
    >
      <span
        className="cell-content"
      >
        <input
          type="checkbox"
        />
      </span>
    </div>
    <div
      className="sc-bdVaJa buROYz cell-id"
    >
      <span
        className="cell-content"
      >
        4217
      </span>
    </div>
    <div
      className="sc-bdVaJa dBnJxf cell-main cell-main-1"
    >
      <span
        className="cell-content"
      >
        Something
      </span>
    </div>
    <div
      className="sc-bdVaJa dWysYI cell-basic cell-basic-1"
    >
      <span
        className="cell-content"
      >
        Something else
      </span>
    </div>
  </div>
  <div
    className="sc-bwzfXH iBHjTe"
  >
    <div
      className="sc-bdVaJa fCCeHr cell-checkbox"
    >
      <span
        className="cell-content"
      >
        <input
          type="checkbox"
        />
      </span>
    </div>
    <div
      className="sc-bdVaJa buROYz cell-id"
    >
      <span
        className="cell-content"
      >
        4217
      </span>
    </div>
    <div
      className="sc-bdVaJa dBnJxf cell-main cell-main-1"
    >
      <span
        className="cell-content"
      >
        Something
      </span>
    </div>
    <div
      className="sc-bdVaJa dWysYI cell-basic cell-basic-1"
    >
      <span
        className="cell-content"
      >
        Something else
      </span>
    </div>
  </div>
  <div
    className="sc-bwzfXH iBHjTe"
  >
    <div
      className="sc-bdVaJa fCCeHr cell-checkbox"
    >
      <span
        className="cell-content"
      >
        <input
          type="checkbox"
        />
      </span>
    </div>
    <div
      className="sc-bdVaJa buROYz cell-id"
    >
      <span
        className="cell-content"
      >
        4217
      </span>
    </div>
    <div
      className="sc-bdVaJa dBnJxf cell-main cell-main-1"
    >
      <span
        className="cell-content"
      >
        Something
      </span>
    </div>
    <div
      className="sc-bdVaJa dWysYI cell-basic cell-basic-1"
    >
      <span
        className="cell-content"
      >
        Something else
      </span>
    </div>
  </div>
</div>
`;
